### PR TITLE
Plugin error logging

### LINKED ISSUE #18: Store Errors From The Plugin
- Allow the plugin to send us errors and debug information so we can debug.

### PR DESCRIPTION
**Pull request by AndyTWF**
Close #18 

Allows the plugin to post us JSON data regarding its state, so that we have debugging information.

### CODE REVIEW

File: app/Http/Controllers/PluginErrorController.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        $pluginError = PluginError::create(
+            [
+                'user_id' => Auth::user()->id,
+                'user_report' => $request->json('user_report'),
+                'data' => json_encode($request->json('data')),
+            ]
+        );
+
+        Event::fire(new PluginErrorReceivedEvent($pluginError));
+        return response('', 204);
```
Comment (CalumTowers) [inline app/Http/Controllers/PluginErrorController.php]: Not sure a 204 makes sense? The response should acknowledge the creation of an event (201) or confirm all is okay (200).

File: app/Listeners/PluginError/RecordPluginErrorInBugsnag.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+{
+    /**
+     * Handle any squawk allocation event
+     *
+     * @param PluginErrorReceivedEvent $errorReceivedEvent
+     * @return bool
+     */
+    public function handle(PluginErrorReceivedEvent $errorReceivedEvent) : bool
+    {
+        Bugsnag::notifyException(new PluginErrorException('New plugin error reported'));
```
Comment (CalumTowers) [inline app/Listeners/PluginError/RecordPluginErrorInBugsnag.php]: This isn't required; Bugsnag will catch any exception that bubbles up to the `ExceptionHandler` class.

File: tests/app/Listeners/PluginError/RecordPluginErrorInBugsnagTest.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Listeners\PluginError;
+
+use App\BaseUnitTestCase;
+use App\Events\PluginErrorReceivedEvent;
+use App\Models\PluginError\PluginError;
+
+class RecordPluginErrorInBugsnagTest extends BaseUnitTestCase
```
Comment (CalumTowers) [inline tests/app/Listeners/PluginError/RecordPluginErrorInBugsnagTest.php]: I'm not sure this test class has much value?

File: app/Http/Controllers/PluginErrorController.php (hunk 1 above)
Comment (AndyTWF) [reply app/Http/Controllers/PluginErrorController.php]: Changed to 201

File: tests/app/Listeners/PluginError/RecordPluginErrorInBugsnagTest.php (hunk 3 above)
Comment (AndyTWF) [reply tests/app/Listeners/PluginError/RecordPluginErrorInBugsnagTest.php]: I'm a stickler for high test coverage, that's why it exists xD

Comment (CalumTowers) [reply tests/app/Listeners/PluginError/RecordPluginErrorInBugsnagTest.php]: If my Bugsnag comment above is addressed it should make this test (and the event) redundant

**Comment by CalumTowers** [comment]
As discussed, we'll likely need to flesh this out a tad more.

**Pull request closed without merging**